sdk/helper/random/rules_test.go

```diff
@@ -0,0 +1,90 @@
+package random
+
+import (
+	"testing"
+)
+
+func TestCharsetRestriction(t *testing.T) {
+	type testCase struct {
+		charset  string
+		minChars int
+		input    string
+		expected bool
+	}
+
+	tests := map[string]testCase{
+		"0 minimum, empty input": {
+			charset:  "abcdefghijklmnopqrstuvwxyz",
```

Comment:
Nit: maybe setup a variable that has this so we don't have subtle errors with these long strings?

Reply:
Done (used the charset variables from your previous comment)